File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        open_cnt = 0
        res = 0
        for c in s:
            if c == '(':
                open_cnt += 1
            elif open_cnt > 0:
                open_cnt -= 1
            else:
                res += 1
        return res + open_cnt

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int ret = 0;
        int cnt = 0;
        for (char x : s.toCharArray()) {
            if (x == '(') {
                cnt++;
            } else if (cnt > 0) {
                cnt--;
            } else {
                ret++;
            }
        }
        return ret + cnt;
    }
}